File: src/main/java/de/synyx/google/calendar/api/service/Query.java
package de.synyx.google.calendar.api.service;

import java.time.ZonedDateTime;
import java.time.temporal.ChronoUnit;
import java.util.stream.Stream;

/**
 * @author clausen - clausen@synyx.de
 */
public interface Query<T> {

    public Stream<T> between  (ZonedDateTime start, ZonedDateTime end);

    public Stream<T> starting (ZonedDateTime start);

    public default Stream<T> day (ZonedDateTime start) {
        ZonedDateTime   startofday = start.truncatedTo (ChronoUnit.DAYS);
        return between (startofday, startofday.plusDays (1));
    }

}

File: src/main/kotlin/de/synyx/google/calendar/api/service/Query.kt
package de.synyx.google.calendar.api.service

import de.synyx.google.calendar.api.model.Event
import org.joda.time.Instant
import java.util.*

/**
 * @author clausen - clausen@synyx.de
 */
interface Query {

    fun between  (start : Instant, end : Instant, zone : TimeZone) : Collection<Event>

    fun starting (start : Instant, zone : TimeZone) : Collection<Event>

}
